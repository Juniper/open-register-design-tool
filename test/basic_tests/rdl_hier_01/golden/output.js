/*   Ordt 190725.01 autogenerated file 
 *   Input: ./rdl_hier_01/test.rdl
 *   Parms: ./rdl_hier_01/test.parms
 *   Date: Sun Nov 17 17:00:58 EST 2019
 */

register_set base_map "L1 regs" {

  address = 0x0;
  register_width = 32;
  register_set base_regs "base_regs register_set" {
    address = 0x0;
    
    register config_regs "Config reg set  config_regs" {
      address = 0x0;
      repeat = 8;
      integer msb_field[16] "Bunch o msb bits" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer lsb_field[16] "Bunch o lsb bits" param {
        access_mode = READ_WRITE;
        reset = 0xffff;
      };
    };
    
    register state_regs "State regs" {
      address = 0x100;
      access_mode = READ_ONLY;
      repeat = 8;
      nop[16];
      integer lsb_field[16] "Some state bits" param {
        access_mode = READ_ONLY;
        reset = 0xffff;
      };
    };
    
    register_set_size = 0x120;
  };
  
  register_set ext_base_regs "ext_base_regs register_set" {
    address = 0x10000;
    
    register config_regs "Config reg set  config_regs" {
      address = 0x0;
      repeat = 8;
      integer msb_field[16] "Bunch o msb bits" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer lsb_field[16] "Bunch o lsb bits" param {
        access_mode = READ_WRITE;
        reset = 0xffff;
      };
    };
    
    register state_regs "State regs" {
      address = 0x100;
      access_mode = READ_ONLY;
      repeat = 8;
      nop[16];
      integer lsb_field[16] "Some state bits" param {
        access_mode = READ_ONLY;
        reset = 0xffff;
      };
    };
    
    register_set_size = 0x200;
  };
  
  register_set l2_r16_child "L2 regs" {
    address = 0x20000;
    
    register_set base_regs "base_regs register_set" {
      address = 0x0;
      
      register config_regs "Config reg set  config_regs" {
        address = 0x0;
        repeat = 8;
        integer msb_field[16] "Bunch o msb bits" param {
          access_mode = READ_WRITE;
          reset = 0x0;
        };
        integer lsb_field[16] "Bunch o lsb bits" param {
          access_mode = READ_WRITE;
          reset = 0xffff;
        };
      };
      
      register state_regs "State regs" {
        address = 0x100;
        access_mode = READ_ONLY;
        repeat = 8;
        nop[16];
        integer lsb_field[16] "Some state bits" param {
          access_mode = READ_ONLY;
          reset = 0xffff;
        };
      };
      
      register_set_size = 0x120;
    };
    
    register_set ext_base_regs "ext_base_regs register_set" {
      address = 0x1000;
      
      register config_regs "Config reg set  config_regs" {
        address = 0x0;
        repeat = 8;
        integer msb_field[16] "Bunch o msb bits" param {
          access_mode = READ_WRITE;
          reset = 0x0;
        };
        integer lsb_field[16] "Bunch o lsb bits" param {
          access_mode = READ_WRITE;
          reset = 0xffff;
        };
      };
      
      register state_regs "State regs" {
        address = 0x100;
        access_mode = READ_ONLY;
        repeat = 8;
        nop[16];
        integer lsb_field[16] "Some state bits" param {
          access_mode = READ_ONLY;
          reset = 0xffff;
        };
      };
      
      register_set_size = 0x200;
    };
    
    register_set l3_child "L3 regs" {
      address = 0x2000;
      
      register_set base_regs "base_regs register_set" {
        address = 0x0;
        
        register config_regs "Config reg set  config_regs" {
          address = 0x0;
          repeat = 8;
          integer msb_field[16] "Bunch o msb bits" param {
            access_mode = READ_WRITE;
            reset = 0x0;
          };
          integer lsb_field[16] "Bunch o lsb bits" param {
            access_mode = READ_WRITE;
            reset = 0xffff;
          };
        };
        
        register state_regs "State regs" {
          address = 0x100;
          access_mode = READ_ONLY;
          repeat = 8;
          nop[16];
          integer lsb_field[16] "Some state bits" param {
            access_mode = READ_ONLY;
            reset = 0xffff;
          };
        };
        
        register_set_size = 0x120;
      };
      
      register_set ext_base_regs "ext_base_regs register_set" {
        address = 0x800;
        
        register config_regs "Config reg set  config_regs" {
          address = 0x0;
          repeat = 8;
          integer msb_field[16] "Bunch o msb bits" param {
            access_mode = READ_WRITE;
            reset = 0x0;
          };
          integer lsb_field[16] "Bunch o lsb bits" param {
            access_mode = READ_WRITE;
            reset = 0xffff;
          };
        };
        
        register state_regs "State regs" {
          address = 0x100;
          access_mode = READ_ONLY;
          repeat = 8;
          nop[16];
          integer lsb_field[16] "Some state bits" param {
            access_mode = READ_ONLY;
            reset = 0xffff;
          };
        };
        
        register_set_size = 0x200;
      };
      
      register_set_size = 0xa00;
    };
    
    register_set_size = 0x4000;
  };
  
  register_set l2_s8_child "L2 regs" {
    address = 0x30000;
    
    register_set base_regs "base_regs register_set" {
      address = 0x0;
      
      register config_regs "Config reg set  config_regs" {
        address = 0x0;
        repeat = 8;
        integer msb_field[16] "Bunch o msb bits" param {
          access_mode = READ_WRITE;
          reset = 0x0;
        };
        integer lsb_field[16] "Bunch o lsb bits" param {
          access_mode = READ_WRITE;
          reset = 0xffff;
        };
      };
      
      register state_regs "State regs" {
        address = 0x100;
        access_mode = READ_ONLY;
        repeat = 8;
        nop[16];
        integer lsb_field[16] "Some state bits" param {
          access_mode = READ_ONLY;
          reset = 0xffff;
        };
      };
      
      register_set_size = 0x120;
    };
    
    register_set ext_base_regs "ext_base_regs register_set" {
      address = 0x1000;
      
      register config_regs "Config reg set  config_regs" {
        address = 0x0;
        repeat = 8;
        integer msb_field[16] "Bunch o msb bits" param {
          access_mode = READ_WRITE;
          reset = 0x0;
        };
        integer lsb_field[16] "Bunch o lsb bits" param {
          access_mode = READ_WRITE;
          reset = 0xffff;
        };
      };
      
      register state_regs "State regs" {
        address = 0x100;
        access_mode = READ_ONLY;
        repeat = 8;
        nop[16];
        integer lsb_field[16] "Some state bits" param {
          access_mode = READ_ONLY;
          reset = 0xffff;
        };
      };
      
      register_set_size = 0x200;
    };
    
    register_set l3_child "L3 regs" {
      address = 0x2000;
      
      register_set base_regs "base_regs register_set" {
        address = 0x0;
        
        register config_regs "Config reg set  config_regs" {
          address = 0x0;
          repeat = 8;
          integer msb_field[16] "Bunch o msb bits" param {
            access_mode = READ_WRITE;
            reset = 0x0;
          };
          integer lsb_field[16] "Bunch o lsb bits" param {
            access_mode = READ_WRITE;
            reset = 0xffff;
          };
        };
        
        register state_regs "State regs" {
          address = 0x100;
          access_mode = READ_ONLY;
          repeat = 8;
          nop[16];
          integer lsb_field[16] "Some state bits" param {
            access_mode = READ_ONLY;
            reset = 0xffff;
          };
        };
        
        register_set_size = 0x120;
      };
      
      register_set ext_base_regs "ext_base_regs register_set" {
        address = 0x800;
        
        register config_regs "Config reg set  config_regs" {
          address = 0x0;
          repeat = 8;
          integer msb_field[16] "Bunch o msb bits" param {
            access_mode = READ_WRITE;
            reset = 0x0;
          };
          integer lsb_field[16] "Bunch o lsb bits" param {
            access_mode = READ_WRITE;
            reset = 0xffff;
          };
        };
        
        register state_regs "State regs" {
          address = 0x100;
          access_mode = READ_ONLY;
          repeat = 8;
          nop[16];
          integer lsb_field[16] "Some state bits" param {
            access_mode = READ_ONLY;
            reset = 0xffff;
          };
        };
        
        register_set_size = 0x200;
      };
      
      register_set_size = 0xa00;
    };
    
    register_set_size = 0x4000;
  };
  
  register no_opt_reg "State reg" {
    address = 0x40000;
    access_mode = READ_ONLY;
    nop[16];
    integer lsb_field[16] "Some state bits" param {
      access_mode = READ_ONLY;
      reset = 0xffff;
    };
  };
  
  register opt_reg "State reg" {
    address = 0x40010;
    access_mode = READ_ONLY;
    nop[16];
    integer lsb_field[16] "Some state bits" param {
      access_mode = READ_ONLY;
      reset = 0xffff;
    };
  };
  
  register_set_size = 0x40014;
};

