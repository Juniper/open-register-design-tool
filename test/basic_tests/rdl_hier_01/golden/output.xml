<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 190725.01 autogenerated file 
       Input: ./rdl_hier_01/test.rdl
       Parms: ./rdl_hier_01/test.parms
       Date: Sun Nov 17 17:08:19 EST 2019
 -->

<map version="190725.01">
  <id>base_map</id>
  <baseaddr>0x0</baseaddr>
  <shorttext>L1 regs</shorttext>
  <regset>
    <id>base_regs</id>
    <shorttext>base_regs registers</shorttext>
    <baseaddr>0x0</baseaddr>
    <reg>
      <id>config_regs</id>
      <parentpath>base_regs</parentpath>
      <shorttext>Config reg set  config_regs</shorttext>
      <baseaddr>0x0</baseaddr>
      <width>32</width>
      <reps>8</reps>
      <stride>0x4</stride>
      <field>
        <id>lsb_field</id>
        <shorttext>Bunch o lsb bits</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>ffff</reset>
        <lowidx>0</lowidx>
        <width>16</width>
      </field>
      <field>
        <id>msb_field</id>
        <shorttext>Bunch o msb bits</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0</reset>
        <lowidx>16</lowidx>
        <width>16</width>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>state_regs</id>
      <parentpath>base_regs</parentpath>
      <shorttext>State regs</shorttext>
      <baseaddr>0x100</baseaddr>
      <width>32</width>
      <reps>8</reps>
      <stride>0x4</stride>
      <field>
        <id>lsb_field</id>
        <shorttext>Some state bits</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
        </hwinfo>
        <reset>ffff</reset>
        <lowidx>0</lowidx>
        <width>16</width>
        <hwmod></hwmod>
      </field>
      <access>RO</access>
    </reg>
    <highaddr>0x11f</highaddr>
  </regset>
  <regset>
    <id>ext_base_regs</id>
    <shorttext>ext_base_regs registers</shorttext>
    <baseaddr>0x10000</baseaddr>
    <extroot>1</extroot>
    <reg>
      <id>config_regs</id>
      <parentpath>ext_base_regs</parentpath>
      <shorttext>Config reg set  config_regs</shorttext>
      <baseaddr>0x10000</baseaddr>
      <width>32</width>
      <reps>8</reps>
      <stride>0x4</stride>
      <field>
        <id>lsb_field</id>
        <shorttext>Bunch o lsb bits</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>ffff</reset>
        <lowidx>0</lowidx>
        <width>16</width>
      </field>
      <field>
        <id>msb_field</id>
        <shorttext>Bunch o msb bits</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0</reset>
        <lowidx>16</lowidx>
        <width>16</width>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>state_regs</id>
      <parentpath>ext_base_regs</parentpath>
      <shorttext>State regs</shorttext>
      <baseaddr>0x10100</baseaddr>
      <width>32</width>
      <reps>8</reps>
      <stride>0x4</stride>
      <field>
        <id>lsb_field</id>
        <shorttext>Some state bits</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
        </hwinfo>
        <reset>ffff</reset>
        <lowidx>0</lowidx>
        <width>16</width>
        <hwmod></hwmod>
      </field>
      <access>RO</access>
    </reg>
    <highaddr>0x1011f</highaddr>
  </regset>
  <map>
    <id>l2_r16_child</id>
    <shorttext>L2 regs</shorttext>
    <baseaddr>0x20000</baseaddr>
    <extroot>1</extroot>
    <regset>
      <id>base_regs</id>
      <parentpath>l2_r16_child</parentpath>
      <shorttext>base_regs registers</shorttext>
      <baseaddr>0x20000</baseaddr>
      <reg>
        <id>config_regs</id>
        <parentpath>l2_r16_child.base_regs</parentpath>
        <shorttext>Config reg set  config_regs</shorttext>
        <baseaddr>0x20000</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Bunch o lsb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
        </field>
        <field>
          <id>msb_field</id>
          <shorttext>Bunch o msb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>0</reset>
          <lowidx>16</lowidx>
          <width>16</width>
        </field>
        <access>RW</access>
      </reg>
      <reg>
        <id>state_regs</id>
        <parentpath>l2_r16_child.base_regs</parentpath>
        <shorttext>State regs</shorttext>
        <baseaddr>0x20100</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Some state bits</shorttext>
          <access>RO</access>
          <hwinfo>
            <hwaccess>w</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
          <hwmod></hwmod>
        </field>
        <access>RO</access>
      </reg>
      <highaddr>0x2011f</highaddr>
    </regset>
    <regset>
      <id>ext_base_regs</id>
      <parentpath>l2_r16_child</parentpath>
      <shorttext>ext_base_regs registers</shorttext>
      <baseaddr>0x21000</baseaddr>
      <reg>
        <id>config_regs</id>
        <parentpath>l2_r16_child.ext_base_regs</parentpath>
        <shorttext>Config reg set  config_regs</shorttext>
        <baseaddr>0x21000</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Bunch o lsb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
        </field>
        <field>
          <id>msb_field</id>
          <shorttext>Bunch o msb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>0</reset>
          <lowidx>16</lowidx>
          <width>16</width>
        </field>
        <access>RW</access>
      </reg>
      <reg>
        <id>state_regs</id>
        <parentpath>l2_r16_child.ext_base_regs</parentpath>
        <shorttext>State regs</shorttext>
        <baseaddr>0x21100</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Some state bits</shorttext>
          <access>RO</access>
          <hwinfo>
            <hwaccess>w</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
          <hwmod></hwmod>
        </field>
        <access>RO</access>
      </reg>
      <highaddr>0x2111f</highaddr>
    </regset>
    <map>
      <id>l3_child</id>
      <parentpath>l2_r16_child</parentpath>
      <shorttext>L3 regs</shorttext>
      <baseaddr>0x22000</baseaddr>
      <regset>
        <id>base_regs</id>
        <parentpath>l2_r16_child.l3_child</parentpath>
        <shorttext>base_regs registers</shorttext>
        <baseaddr>0x22000</baseaddr>
        <reg>
          <id>config_regs</id>
          <parentpath>l2_r16_child.l3_child.base_regs</parentpath>
          <shorttext>Config reg set  config_regs</shorttext>
          <baseaddr>0x22000</baseaddr>
          <width>32</width>
          <reps>8</reps>
          <stride>0x4</stride>
          <field>
            <id>lsb_field</id>
            <shorttext>Bunch o lsb bits</shorttext>
            <access>RW</access>
            <hwinfo>
              <hwaccess>r</hwaccess>
            </hwinfo>
            <reset>ffff</reset>
            <lowidx>0</lowidx>
            <width>16</width>
          </field>
          <field>
            <id>msb_field</id>
            <shorttext>Bunch o msb bits</shorttext>
            <access>RW</access>
            <hwinfo>
              <hwaccess>r</hwaccess>
            </hwinfo>
            <reset>0</reset>
            <lowidx>16</lowidx>
            <width>16</width>
          </field>
          <access>RW</access>
        </reg>
        <reg>
          <id>state_regs</id>
          <parentpath>l2_r16_child.l3_child.base_regs</parentpath>
          <shorttext>State regs</shorttext>
          <baseaddr>0x22100</baseaddr>
          <width>32</width>
          <reps>8</reps>
          <stride>0x4</stride>
          <field>
            <id>lsb_field</id>
            <shorttext>Some state bits</shorttext>
            <access>RO</access>
            <hwinfo>
              <hwaccess>w</hwaccess>
            </hwinfo>
            <reset>ffff</reset>
            <lowidx>0</lowidx>
            <width>16</width>
            <hwmod></hwmod>
          </field>
          <access>RO</access>
        </reg>
        <highaddr>0x2211f</highaddr>
      </regset>
      <regset>
        <id>ext_base_regs</id>
        <parentpath>l2_r16_child.l3_child</parentpath>
        <shorttext>ext_base_regs registers</shorttext>
        <baseaddr>0x22800</baseaddr>
        <reg>
          <id>config_regs</id>
          <parentpath>l2_r16_child.l3_child.ext_base_regs</parentpath>
          <shorttext>Config reg set  config_regs</shorttext>
          <baseaddr>0x22800</baseaddr>
          <width>32</width>
          <reps>8</reps>
          <stride>0x4</stride>
          <field>
            <id>lsb_field</id>
            <shorttext>Bunch o lsb bits</shorttext>
            <access>RW</access>
            <hwinfo>
              <hwaccess>r</hwaccess>
            </hwinfo>
            <reset>ffff</reset>
            <lowidx>0</lowidx>
            <width>16</width>
          </field>
          <field>
            <id>msb_field</id>
            <shorttext>Bunch o msb bits</shorttext>
            <access>RW</access>
            <hwinfo>
              <hwaccess>r</hwaccess>
            </hwinfo>
            <reset>0</reset>
            <lowidx>16</lowidx>
            <width>16</width>
          </field>
          <access>RW</access>
        </reg>
        <reg>
          <id>state_regs</id>
          <parentpath>l2_r16_child.l3_child.ext_base_regs</parentpath>
          <shorttext>State regs</shorttext>
          <baseaddr>0x22900</baseaddr>
          <width>32</width>
          <reps>8</reps>
          <stride>0x4</stride>
          <field>
            <id>lsb_field</id>
            <shorttext>Some state bits</shorttext>
            <access>RO</access>
            <hwinfo>
              <hwaccess>w</hwaccess>
            </hwinfo>
            <reset>ffff</reset>
            <lowidx>0</lowidx>
            <width>16</width>
            <hwmod></hwmod>
          </field>
          <access>RO</access>
        </reg>
        <highaddr>0x2291f</highaddr>
      </regset>
      <highaddr>0x229ff</highaddr>
    </map>
    <highaddr>0x229ff</highaddr>
  </map>
  <map>
    <id>l2_s8_child</id>
    <shorttext>L2 regs</shorttext>
    <baseaddr>0x30000</baseaddr>
    <extroot>1</extroot>
    <regset>
      <id>base_regs</id>
      <parentpath>l2_s8_child</parentpath>
      <shorttext>base_regs registers</shorttext>
      <baseaddr>0x30000</baseaddr>
      <reg>
        <id>config_regs</id>
        <parentpath>l2_s8_child.base_regs</parentpath>
        <shorttext>Config reg set  config_regs</shorttext>
        <baseaddr>0x30000</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Bunch o lsb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
        </field>
        <field>
          <id>msb_field</id>
          <shorttext>Bunch o msb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>0</reset>
          <lowidx>16</lowidx>
          <width>16</width>
        </field>
        <access>RW</access>
      </reg>
      <reg>
        <id>state_regs</id>
        <parentpath>l2_s8_child.base_regs</parentpath>
        <shorttext>State regs</shorttext>
        <baseaddr>0x30100</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Some state bits</shorttext>
          <access>RO</access>
          <hwinfo>
            <hwaccess>w</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
          <hwmod></hwmod>
        </field>
        <access>RO</access>
      </reg>
      <highaddr>0x3011f</highaddr>
    </regset>
    <regset>
      <id>ext_base_regs</id>
      <parentpath>l2_s8_child</parentpath>
      <shorttext>ext_base_regs registers</shorttext>
      <baseaddr>0x31000</baseaddr>
      <reg>
        <id>config_regs</id>
        <parentpath>l2_s8_child.ext_base_regs</parentpath>
        <shorttext>Config reg set  config_regs</shorttext>
        <baseaddr>0x31000</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Bunch o lsb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
        </field>
        <field>
          <id>msb_field</id>
          <shorttext>Bunch o msb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>0</reset>
          <lowidx>16</lowidx>
          <width>16</width>
        </field>
        <access>RW</access>
      </reg>
      <reg>
        <id>state_regs</id>
        <parentpath>l2_s8_child.ext_base_regs</parentpath>
        <shorttext>State regs</shorttext>
        <baseaddr>0x31100</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Some state bits</shorttext>
          <access>RO</access>
          <hwinfo>
            <hwaccess>w</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
          <hwmod></hwmod>
        </field>
        <access>RO</access>
      </reg>
      <highaddr>0x3111f</highaddr>
    </regset>
    <map>
      <id>l3_child</id>
      <parentpath>l2_s8_child</parentpath>
      <shorttext>L3 regs</shorttext>
      <baseaddr>0x32000</baseaddr>
      <regset>
        <id>base_regs</id>
        <parentpath>l2_s8_child.l3_child</parentpath>
        <shorttext>base_regs registers</shorttext>
        <baseaddr>0x32000</baseaddr>
        <reg>
          <id>config_regs</id>
          <parentpath>l2_s8_child.l3_child.base_regs</parentpath>
          <shorttext>Config reg set  config_regs</shorttext>
          <baseaddr>0x32000</baseaddr>
          <width>32</width>
          <reps>8</reps>
          <stride>0x4</stride>
          <field>
            <id>lsb_field</id>
            <shorttext>Bunch o lsb bits</shorttext>
            <access>RW</access>
            <hwinfo>
              <hwaccess>r</hwaccess>
            </hwinfo>
            <reset>ffff</reset>
            <lowidx>0</lowidx>
            <width>16</width>
          </field>
          <field>
            <id>msb_field</id>
            <shorttext>Bunch o msb bits</shorttext>
            <access>RW</access>
            <hwinfo>
              <hwaccess>r</hwaccess>
            </hwinfo>
            <reset>0</reset>
            <lowidx>16</lowidx>
            <width>16</width>
          </field>
          <access>RW</access>
        </reg>
        <reg>
          <id>state_regs</id>
          <parentpath>l2_s8_child.l3_child.base_regs</parentpath>
          <shorttext>State regs</shorttext>
          <baseaddr>0x32100</baseaddr>
          <width>32</width>
          <reps>8</reps>
          <stride>0x4</stride>
          <field>
            <id>lsb_field</id>
            <shorttext>Some state bits</shorttext>
            <access>RO</access>
            <hwinfo>
              <hwaccess>w</hwaccess>
            </hwinfo>
            <reset>ffff</reset>
            <lowidx>0</lowidx>
            <width>16</width>
            <hwmod></hwmod>
          </field>
          <access>RO</access>
        </reg>
        <highaddr>0x3211f</highaddr>
      </regset>
      <regset>
        <id>ext_base_regs</id>
        <parentpath>l2_s8_child.l3_child</parentpath>
        <shorttext>ext_base_regs registers</shorttext>
        <baseaddr>0x32800</baseaddr>
        <reg>
          <id>config_regs</id>
          <parentpath>l2_s8_child.l3_child.ext_base_regs</parentpath>
          <shorttext>Config reg set  config_regs</shorttext>
          <baseaddr>0x32800</baseaddr>
          <width>32</width>
          <reps>8</reps>
          <stride>0x4</stride>
          <field>
            <id>lsb_field</id>
            <shorttext>Bunch o lsb bits</shorttext>
            <access>RW</access>
            <hwinfo>
              <hwaccess>r</hwaccess>
            </hwinfo>
            <reset>ffff</reset>
            <lowidx>0</lowidx>
            <width>16</width>
          </field>
          <field>
            <id>msb_field</id>
            <shorttext>Bunch o msb bits</shorttext>
            <access>RW</access>
            <hwinfo>
              <hwaccess>r</hwaccess>
            </hwinfo>
            <reset>0</reset>
            <lowidx>16</lowidx>
            <width>16</width>
          </field>
          <access>RW</access>
        </reg>
        <reg>
          <id>state_regs</id>
          <parentpath>l2_s8_child.l3_child.ext_base_regs</parentpath>
          <shorttext>State regs</shorttext>
          <baseaddr>0x32900</baseaddr>
          <width>32</width>
          <reps>8</reps>
          <stride>0x4</stride>
          <field>
            <id>lsb_field</id>
            <shorttext>Some state bits</shorttext>
            <access>RO</access>
            <hwinfo>
              <hwaccess>w</hwaccess>
            </hwinfo>
            <reset>ffff</reset>
            <lowidx>0</lowidx>
            <width>16</width>
            <hwmod></hwmod>
          </field>
          <access>RO</access>
        </reg>
        <highaddr>0x3291f</highaddr>
      </regset>
      <highaddr>0x329ff</highaddr>
    </map>
    <highaddr>0x329ff</highaddr>
  </map>
  <reg>
    <id>no_opt_reg</id>
    <shorttext>State reg</shorttext>
    <baseaddr>0x40000</baseaddr>
    <width>32</width>
    <extroot></extroot>
    <field>
      <id>lsb_field</id>
      <shorttext>Some state bits</shorttext>
      <access>RO</access>
      <hwinfo>
        <hwaccess>w</hwaccess>
      </hwinfo>
      <reset>ffff</reset>
      <lowidx>0</lowidx>
      <width>16</width>
      <hwmod></hwmod>
    </field>
    <access>RO</access>
  </reg>
  <reg>
    <id>opt_reg</id>
    <shorttext>State reg</shorttext>
    <baseaddr>0x40010</baseaddr>
    <width>32</width>
    <extroot></extroot>
    <field>
      <id>lsb_field</id>
      <shorttext>Some state bits</shorttext>
      <access>RO</access>
      <hwinfo>
        <hwaccess>w</hwaccess>
      </hwinfo>
      <reset>ffff</reset>
      <lowidx>0</lowidx>
      <width>16</width>
      <hwmod></hwmod>
    </field>
    <access>RO</access>
  </reg>
</map>
